### DISCUSSION TITLE
Running Horreum app on s390x architecture

### DISCUSSION THREAD

**Discussion by ShivangGoswami** [Hyperfoil/Horreum#338] [Q&A]
Previous thread: https://github.com/Hyperfoil/Horreum/issues/329

1. Tried https://github.com/quarkusio/quarkus-images for running horreum app 
   - [Quay images](https://quay.io/organization/quarkus) which are available only for amd64 & arm64 architect only
   - Tried from build these images from source but these are dependency on [mandrel ](https://github.com/quarkusio/quarkus-images/blob/main/quarkus-mandrel-builder-image/mandrel.yaml)& [graalvm](https://github.com/quarkusio/quarkus-images/blob/main/quarkus-graalvm-builder-image/graalvm.yaml) binaries. So was unable to do so.
2. Building from [Mandrel Source Repo](https://github.com/graalvm/mandrel) 
    - [Mandrel Packaging ](https://github.com/graalvm/mandrel-packaging) contains instructions to build from source but has a dependency on [mx](https://github.com/graalvm/mx) & [mandrel](https://github.com/graalvm/mandrel).mx is not supported in s390x architecture Issue is created for same https://github.com/graalvm/mx/issues/272. This will take some more time.
3.  Use a s390x base image and run Horreum in Java mode - need more information in this regards
     - Running these commands 
     `mvn -DskipTests=true package`
     `mvn -Dquarkus.test.continuous-testing=disabled quarkus:dev`
      launches an interactive shell.How can disable the interactive nature here?

please help me with point 3

**Comment by jesperpedersen** [comment]
Sorry, I don't have access to s390x, so you will have to drive this.

The Quarkus Maven plugin is documented as 

 https://quarkus.io/guides/quarkus-maven-plugin

You don't have to run it in development mode. What are you looking to do ?

**Comment by ShivangGoswami** [comment]
my intent here is execute the horreum app inside a podman container.

the default build or [quay image](https://quay.io/repository/hyperfoil/horreum?tab=tags) gives error when executing in s390x environment.
for native build, there are errors when compiling the code which needs support from communities to provide s390x based executable enablement.

Just wanted to know that is there any method by which we can run a horreum app inside a s390x environment, it might not be production ready/production enabled env for now. so that we are unblocked with our work.
Horreum is a part of performance testing framework which we are migrating to s390x arch for now.

**Comment by jesperpedersen** [comment]
You probably want a standalone distribution right now.

I have created #339 as an issue for that.

Do you want to work on it ? You can use the sister project https://github.com/Hyperfoil/Hyperfoil/tree/master/distribution files as a template.

**Comment by stalep** [comment]
Closing this for now. We can not prioritize Horreum on s390x environment atm, but we will try to get Hyperfoil working on non-intel platforms in the future.
